tgui/packages/tgui/interfaces/NtosMessenger.tsx

```diff
@@ -0,0 +1,469 @@
+import { useBackend, useLocalState } from '../backend';
+import { createSearch } from 'common/string';
+import { Box, Button, Dimmer, Icon, Section, Stack, Input, TextArea } from '../components';
+import { NtosWindow } from '../layouts';
+import { Component, createRef, InfernoNode, RefObject, SFC } from 'inferno';
+import { sanitizeText } from '../sanitize';
+
+import '../styles/interfaces/NtosMessenger.scss';
+
+type NtMessage = {
+  name: string;
+  job: string;
+  contents: string;
+  outgoing: boolean;
+  sender: string;
+  automated: boolean;
+  photo_path: string;
+  photo: string;
+  everyone: boolean;
+  targets: string[];
+  target_details: string[];
+};
+
+type NtMessenger = {
+  name: string;
+  job: string;
+  ref: string;
+};
+
+type NtChat = {
+  messages:
+};
+
+type NtMessengers = Record<string, NtMessenger>;
+
+type NtosMessengerData = {
+  owner: string;
+  sort_by_job: boolean;
```

Comment:
we don't have actual booleans (too new of technology for Byond, which sends 1 and 0), please import & use `BooleanLike`